### PR TITLE
[ML-202] Update oneAPI Base Toolkit version and prepare for OAP 1.3.1 release

### PR DESCRIPTION
**Pull request by HongW2019**
## What changes were proposed in this pull request?

Update oneAPI Base Toolkit version to mitigate vulnerabilities scanned from libmpi and prepare for OAP 1.3.1 release

## Does this PR also require the following changes?

- CI
- Documentation
- Example

### CODE REVIEW

**Comment by HongW2019** [comment]
@xwu99 Please take a review, thanks.

**Comment by xwu99** [comment]
> @xwu99 Please take a review, thanks.

Thanks for your work! 
Checked CI,  dev/prepare-build-deps.sh should also be fixed.
Should also change the version in main README.

**Comment by xwu99** [comment]
@HongW2019 I adjust compiler setting to use gcc for cpu profile and dpcpp for cpu_gpu profile. The old setting of using dpcpp will bring gpu dependencies even for cpu only package.

File: .github/workflows/ci-checks-build.yml [hunk 1]
```diff
@@ -60,7 +60,6 @@ jobs:
           path: |
             # /var/cache/apt/archives/*.deb
             ~/.m2/repository
-            /opt/intel/oneapi
             ~/opt
           key: ${{ runner.os }}_spark-3.2.0_hadoop-3.2.0_oneapi-2021.4.0
```
Comment (xwu99) [inline .github/workflows/ci-checks-build.yml:64]: you don't need to remove /opt/intel/oneapi, just need to change key name(${{ runner.os }}_spark-3.2.0_hadoop-3.2.0_oneapi-2021.4.0), same below

**Review by xwu99 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-checks-build.yml (hunk 1 above)
Comment (HongW2019) [reply .github/workflows/ci-checks-build.yml]: Done.

**Comment by HongW2019** [comment]
> @HongW2019 I adjust compiler setting to use gcc for cpu profile and dpcpp for cpu_gpu profile. The old setting of using dpcpp will bring gpu dependencies even for cpu only package.

Great!  thanks for your help.

**Comment by HongW2019** [comment]
@xwu99 If there is no more action needed for this PR, could you please help merge it? and also help backport those commits that mllib-1.3.1 needs to branch-1.3.1? Thanks a lot.

**Comment by xwu99** [comment]
> @xwu99 If there is no more action needed for this PR, could you please help merge it? and also help backport those commits that mllib-1.3.1 needs to branch-1.3.1? Thanks a lot.

Did you scan the package to see if it fixed the issue #202 ? For 1.3.1, just need to merge all commits since the cutoff. I can do it after this PR merged.

**Comment by HongW2019** [comment]
> > @xwu99 If there is no more action needed for this PR, could you please help merge it? and also help backport those commits that mllib-1.3.1 needs to branch-1.3.1? Thanks a lot.
> 
> Did you scan the package to see if it fixed the issue #202 ? For 1.3.1, just need to merge all commits since the cutoff. I can do it after this PR merged.

I have locally tested this PR before `11960fd7b07` committed, it can solve the the json-c vulnerability brought from libmpi.

Now to build oap-mllib with the whole patch locally, the package can also avoid the the json-c vulnerability brought from libmpi.  I also found that building env had limit to gcc version after `11960fd7b07` committed. In my previous build env, gcc 7.3 worked good on CentOS 7, but didn't work for current whole PR to build. I tried gcc 9 then oap-mllib was built successfully.

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
             ~/.m2/repository
             /opt/intel/oneapi
             ~/opt
-          key: ${{ runner.os }}_spark-3.2.0_hadoop-3.2.0_oneapi-2021.4.0
+          key: ${{ runner.os }}_spark-3.2.0_hadoop-3.2.0_oneapi-2021.1.2
```
Comment (xwu99) [inline .github/workflows/ci-tests.yml:47]:
```suggestion
          key: ${{ runner.os }}_spark-3.2.0_hadoop-3.2.0_oneapi-2022.1.2
```

**Comment by xwu99** [comment]
> 

What is your build issue? My env is gcc 7 in ubuntu 18.04.  It worked perfectly.

**Comment by HongW2019** [comment]
> > 
> 
> Previously the compiler is clang from oneapi package. it doesn't use system gcc. What is your build issue right now? My env is gcc 7 in ubuntu 18.04. It worked perfectly for this patch. You can also check CI "OAP MLlib Checks and Build". It's exactly the same with my local result.

You are right, gcc 7 works well to build oap-mllib. I use another clean machine with gcc 7 to build and it works, so previously what I mentioned isn't an issue.

**Comment by xwu99** [comment]
@HongW2019 Thanks for your work! merged for master and 1.3.1 branch

**Pull request merged**